test.js

```diff
@@ -17,6 +17,24 @@ test('memoize', t => {
 	t.is(memoized('foo', 'bar'), 2);
 });
 
+test('memoize itself is memoized', t => {
+	let i = 0;
+	const f = () => i++;
+	let memoized = m(f);
+	t.is(memoized(), 0);
+	t.is(memoized(), 0);
+	memoized = m(f);
+	t.is(memoized(), 0);
+	t.is(memoized('foo'), 1);
+	t.is(memoized('foo'), 1);
+	memoized = m(f);
+	t.is(memoized('foo'), 1);
+	t.is(memoized('foo', 'bar'), 2);
+	t.is(memoized('foo', 'bar'), 2);
+	memoized = m(f);
+	t.is(memoized('foo', 'bar'), 2);
+});
```

Comment:
Needs some more tests using different options, especially testing with the `cache` option.